clients/rospy/src/rospy/topics.py

```diff
@@ -1266,7 +1267,11 @@ def acquire_impl(self, reg_type, resolved_name, data_class):
         else:
             raise TypeError("invalid reg_type: %s"%reg_type)
         with self.lock:
-            impl = rmap.get(resolved_name, None)            
+            impl = rmap.get(resolved_name, None)
+            if impl and reg_type == Registration.SUB:
+                _rosout_logger.warning("more than one subscription to %s - "
+                    "subscribing twice to the same topic in one process is not "
```

Comment:
TODO: only if requested types are different (see #1351), e.g. AnyMsg and LaserScan